### PR TITLE
feat: System and appservice module updates

### PR DESCRIPTION
**Pull request by badboy-huaqiao**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/edgex-ui-go/blob/master/.github/Contributing.md.

## What is the current behavior?

hard-code app service list and system service list

## What is the new behavior?

- Add backend support for achieving all app services and system services with new method.
- Update whole app service module, removed custom add new app function, and don't support delete app in the frontend any more.
- Update whole system agent module, using backend server to achieving all registered service from register center instead of hard-code.
- Other updates like remove dead code in system agent module, add associated contracts model and service.

## Does this PR introduce a breaking change?

- [ ] Yes
- [* ] No

## New Imports

- [ ] Yes
- [ *] No

## Specific Instructions
Are there any specific instructions or things that should be known prior to reviewing?

## Other information

### CODE REVIEW

**Review by bnevis-i — DISMISSED** [review]
Should be "user-defined secrets" but I'm not going to be picky.

**Comment by badboy-huaqiao** [comment]
@bnevis-i  sorry, what's "user-defined secrets" exactly mean? I'm not clear about that, it's ok it you think there is a better way, I can update that.

**Comment by bnevis-i** [comment]
> @bnevis-i sorry, what's "user-defined secrets" exactly mean? I'm not clear about that, it's ok if you think there is a better way, I can update that.

Had multiple windows open and commented on the wrong PR.

File: web/src/app/app-service/app-service-configurable/insecure-secrets/insecure-secrets.component.html [hunk 1]
```diff
@@ -17,6 +17,12 @@
  -->
 
  <div *ngIf="insecureSecrets">
+    <div class="alert alert-info mb-3" role="alert">
+      <small>
+          <i class="fa fa-info-circle"></i> 
+          Insecure Secrets that are used when running in non-secure mode, i.e. when Vault isn't available. This is a dynamic map of configuration, so can empty if no secrets are used or can have as many or few user define secrets. It simulates a Secret Store in non-secure mode.
```
Comment (bnevis-i) [inline web/src/app/app-service/app-service-configurable/insecure-secrets/insecure-secrets.component.html:23]:
```suggestion
          Insecure Secrets that are used when running in non-secure mode, i.e. when Vault isn't available. This is a dynamic map of configuration, so can empty if no secrets are used or can have as many or few user-defined secrets. It simulates a Secret Store in non-secure mode. 
```

File: web/src/app/app-service/app-service-list/app-service-list.component.html [hunk 2]
```diff
@@ -21,11 +21,11 @@
          <i class="fa fa-list mr-2 text-danger"></i>
          <span class="font-weight-bold">App Service List</span>
      </div>
-     <div class="card-header py-2">
+     <!-- <div class="card-header py-2">
          <a class="btn btn-sm btn-primary" routerLink="../add-app-service">
              <i class="fa fa-plus mr-2"></i>New App
          </a>
-     </div>
+     </div> -->
```
Comment (bnevis-i) [inline web/src/app/app-service/app-service-list/app-service-list.component.html:28]: Why not just remove it?

File: web/src/app/app-service/app-service-list/app-service-list.component.html [hunk 3]
```diff
@@ -36,14 +36,14 @@
                     <div class="media-body">
                         <div class="mt-0 mb-3">
                             <h5 class="d-inline mr-3 align-middle text-dark">
-                                {{appSvc.name}}
-                                <i class="fa fa-trash text-danger float-right" role="button" (click)="removeSvc(appSvc)"></i>
+                                {{appSvc.ServiceId}}
+                                <!-- <i class="fa fa-trash text-danger float-right" role="button" (click)="removeSvc(appSvc)"></i> -->
```
Comment (bnevis-i) [inline web/src/app/app-service/app-service-list/app-service-list.component.html:40]: Ditto.

**Comment by badboy-huaqiao** [comment]
> > @bnevis-i sorry, what's "user-defined secrets" exactly mean? I'm not clear about that, it's ok if you think there is a better way, I can update that.
> 
> Had multiple windows open and commented on the wrong PR.

do you mean that  there is too many commits in one  PR? so what should I do?

**Review by bnevis-i — DISMISSED** [review]
Approved w/o testing.

Does edgex-ui have a JS UI testing framework integrated like Karma?  Noticed the new code is missing unit tests.

**Comment by bnevis-i** [comment]
> > > @bnevis-i sorry, what's "user-defined secrets" exactly mean? I'm not clear about that, it's ok if you think there is a better way, I can update that.
> > 
> > 
> > Had multiple windows open and commented on the wrong PR.
> 
> do you mean that there are too many commits in one PR? so what should I do?

Actually it was the right review.  Just just thought it was a single-line change.  While formulating the PR, it is often helpful to do "git commit --amend -s -S; git push --force" so that when it goes up for review the 1st time there is only one commit.

File: web/src/app/app-service/app-service-configurable/insecure-secrets/insecure-secrets.component.html (hunk 1 above)
Comment (badboy-huaqiao) [reply web/src/app/app-service/app-service-configurable/insecure-secrets/insecure-secrets.component.html:23]: this messaga is copied from docs.edgexfoundry.org, but I can update it if you think it's not right

File: web/src/app/app-service/app-service-list/app-service-list.component.html (hunk 2 above)
Comment (badboy-huaqiao) [reply web/src/app/app-service/app-service-list/app-service-list.component.html:28]: I'll just comment it out for the possibility of using it again

File: web/src/app/app-service/app-service-configurable/insecure-secrets/insecure-secrets.component.html (hunk 1 above)
Comment (bnevis-i) [reply web/src/app/app-service/app-service-configurable/insecure-secrets/insecure-secrets.component.html:23]: Your PR. Your decision.  I didn't block the merge.

**Comment by badboy-huaqiao** [comment]
> Approved w/o testing.
> 
> Does edgex-ui have a JS UI testing framework integrated like Karma? Noticed the new code is missing unit tests.

well, Karma is a good tools for testing, but I'm not a professional front-end engineer, I am not familiar with how to use Karma, So I need a professional front-end engineer, like you, who can help me do that, and right now,  I'm the only one working on the whole GUI project, I'm very happy working with you.

**Comment by badboy-huaqiao** [comment]
> > > > @bnevis-i sorry, what's "user-defined secrets" exactly mean? I'm not clear about that, it's ok if you think there is a better way, I can update that.
> > > 
> > > 
> > > Had multiple windows open and commented on the wrong PR.
> > 
> > 
> > do you mean that there are too many commits in one PR? so what should I do?
> 
> Actually it was the right review. Just just thought it was a single-line change. While formulating the PR, it is often helpful to do "git commit --amend -s -S; git push --force" so that when it goes up for review the 1st time there is only one commit.

great thanks, your suggestions are really helpful to me, I'm very glad working with you, next time, I will follow your tips to submit a PR.

**Comment by badboy-huaqiao** [comment]
Hi @bnevis-i , I'm really glad to get your advice, that's really helpful to me, I have updated it again, sorry for my poor English, please review that new updates.

And after I discussed with my leader, we are going to do some research about supporting UI testing framework like Karma you mentioned above, please give me a litte more time. thanks again.

**Review by bnevis-i — APPROVED** [review]

**Pull request merged**